### PR TITLE
Update README and post-commit to use asf-staging

### PR DESCRIPTION
**Pull request by ctubbsii**
This is an update to the README, which updates the workflow (and build script) to use a staging branch, and simplifies some git instructions while generating the website. This represents an updated workflow that involves a staging repository, at least until the automated Jekyll builds are set up and working properly (see request at https://issues.apache.org/jira/browse/INFRA-19737).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,47 +1,115 @@
...
+may not be needed.
+
+Because we use [Bundler] to install specific versions of gems, it is not
+recommended to use an OS packaged version of gems other than what comes
+built-in. If you are using an OS packaged version of Ruby, it is __strongly__
+recommended to avoid `sudo` when installing additional gems, in order to avoid
+conflicting with your system's package-managed installation. Instead, you can
+specify a `GEM_HOME` directory for installing gems locally in your home
+directory. You can do this in your `$HOME/.bashrc` file or other appropriate
+place for your environment:
```
Comment (milleruntime) [inline README.md]: Nice!  This burned me bad

File: README.md [hunk 2]
```diff
@@ -1,47 +1,115 @@
...
+## Running Locally for Testing/Development
+
+Rather than merely build the site contents, you can run a local webserver to
+view the contents in your browser. Jekyll provides a simple way to build and
+view the site contents with its own built-in webserver.
+
+Serving the site content this way may result in a view that is slightly
+different than a production site, because Jekyll's embedded webserver may have
+slightly different behavior than the webserver at apache.org, but it should be
+similar enough for testing most things.
```
Comment (milleruntime) [inline README.md]: These two paragraphs are a bit wordy and could be shortened to:
"Jekyll provides a built-in webserver that is sufficient enough for testing but may result in a view that is slightly different than what actually gets deployed at apache.org.

File: README.md [hunk 3]
```diff
@@ -53,54 +121,106 @@ The source for these tags is at [_plugins/links.rb](_plugins/links.rb).
...
+
+# Step 1: update your upstream remote
+git remote update upstream
+
+# Step 2: ensure your local master branch is up-to-date
+# Note: this will destroy any local changes; create a branch first if you want to keep any of that
+git reset --hard upstream/master
+
+# Step 3: create/update your local asf-staging branch
+git branch --track --force asf-staging upstream/asf-staging
```
Comment (milleruntime) [inline README.md]: What is the benefit of using asf-staging?  This seems to complicate the process of updating the website, especially for small changes.  I guess this would be useful for larger changes like re-organizing pages, changing layout, etc.

Comment (ctubbsii) [reply README.md]: The main benefit for now is that you can test the change before you publish it. In future, this benefit may be more obvious when the automated Jekyll building-to-staged stuff is working, at which point, most of this workflow goes away and becomes a simple 3 step process:

1. merge PR in GitHub UI,
2. view automatically staged site to confirm the change (https://accumulo.staged.apache.org),
3. execute simple git command to update asf-site from asf-staging: `git remote update && git push upstream refs/remotes/upstream/asf-staging:asf-site`

Comment (milleruntime) [reply README.md]: > In future, this benefit may be more obvious when the automated Jekyll building-to-staged stuff is working, at which point, most of this workflow goes away and becomes a simple 3 step process

Perhaps we should just wait until then to make changes to the workflow.

Comment (ctubbsii) [reply README.md]: I can modify this to use asf-site again instead of asf-staging for now.

File: README.md (hunk 2 above)
Comment (ctubbsii) [reply README.md]: Will do.

File: README.md (hunk 3 above)
Comment (ctubbsii) [reply README.md]: Rather than modify this to use asf-site, I'm going to cut this entire section out, since I'm working with INFRA on https://issues.apache.org/jira/browse/INFRA-19737 to try to get us set up to automatically build the site when we merge to master. When fully working, the only thing we'll need to do in git is copy the commits in asf-staging to asf-site when we're happy the site is staged correctly.

File: README.md [hunk 4]
```diff
@@ -53,54 +26,139 @@ The source for these tags is at [_plugins/links.rb](_plugins/links.rb).
...
+cp ./core/target/generated-docs/client-properties.md "$ACCUMULO_SITE_CLONE"/_docs-2/configuration
 ```
 
-## Update the production website
+## Local Builds for Testing
+
+### Setting up Your Ruby/Bundler Environment
+
+Ruby and RubyGems are required to use Jekyll and Bundler, so first make sure
+you have those on your machine.
```
Comment (milleruntime) [inline README.md]: It sounds like Ruby and RubyGems are a prerequisite to these instructions instead of the goal. I think something like this would be better:
"Ruby and RubyGems are required to use Jekyll and Bundler.  Follow these guidelines to install each of the pieces on your machine."

Comment (ctubbsii) [reply README.md]: Ruby and RubyGems **are** a prerequisite. The next paragraph talks about the `-devel` packages and other tools that you might also need, but the instructions do not cover installing ruby or rubygems themselves.

Comment (ctubbsii) [reply README.md]: I updated the section header to clarify that the goal is to install the prerequisites for Jekyll/Bundler.

**Comment by ctubbsii** [comment]
NOTE: the information in this README about automatic staging isn't quite working the way it is supposed to yet. I'm tracking the issue at https://issues.apache.org/jira/browse/INFRA-19737

**Comment by ctubbsii** [comment]
Everything seems working great. I sent an email to the dev list to summarize and provide links to the README and other details.
